crates/nu-cli/src/completions/completer.rs

```diff
@@ -142,6 +142,9 @@ impl NuCompleter {
                         let mut spans: Vec<String> = vec![];
 
                         for (flat_idx, flat) in flattened.iter().enumerate() {
+                            let is_sudo_command = !spans.is_empty()
+                                && spans.first().expect("Span shouldn't be empty")
+                                    == &String::from("sudo");
```

Comment:
This `.expect()` feels like danger zone just browsing through the patch. We have still a bunch of panics to track down with completions so I don't feel like that is the right way to go about this line.

Something along the lines of `.filter()`/`.map()`/`Option::contains()` would get you there without trying to check and promise you can unwrap it.
(Probably needs some `Deref` or else to get the types right)

```
                            let is_sudo_command =spans.first().filter(|content| content == "sudo").is_some();
```